XamarinCommunityToolkit/Views/SideMenuView.shared.cs

```diff
@@ -0,0 +1,552 @@
+{
+    [ContentProperty(nameof(Children))]
+    public class SideMenuView : TemplatedView
+    {
+        const string animationName = nameof(SideMenuView);
+
+        const uint animationRate = 16;
+
+        const uint animationLength = 350;
+
+        const int maxTimeDiffItemsCount = 24;
+
+        const int minSwipeTimeDiffItemsCount = 2;
+
+        const double swipeThresholdDistance = 17;
+
+        const double acceptMoveThresholdPercentage = 0.3;
+
+        const uint swipeAnimationAccelerationFactor = 2;
+
+        static readonly Easing animationEasing = Easing.SinOut;
+
+        static readonly TimeSpan swipeThresholdTime = TimeSpan.FromMilliseconds(Device.RuntimePlatform == Device.Android ? 100 : 60);
+
+        public static readonly BindableProperty DiffProperty
+            = BindableProperty.Create(nameof(Diff), typeof(double), typeof(SideMenuView), 0.0, BindingMode.OneWayToSource);
+
+        public static readonly BindableProperty CurrentGestureDiffProperty
+            = BindableProperty.Create(nameof(CurrentGestureDiff), typeof(double), typeof(SideMenuView), 0.0, BindingMode.OneWayToSource);
+
+        public static readonly BindableProperty GestureThresholdProperty
+            = BindableProperty.Create(nameof(GestureThreshold), typeof(double), typeof(SideMenuView), 7.0);
+
+        public static readonly BindableProperty CancelVerticalGestureThresholdProperty
+            = BindableProperty.Create(nameof(CancelVerticalGestureThreshold), typeof(double), typeof(SideMenuView), 1.0);
+
+        public static readonly BindableProperty AllowInterceptGestureProperty
+            = BindableProperty.Create(nameof(AllowInterceptGesture), typeof(bool), typeof(SideMenuView), false);
+
+        public static readonly BindableProperty StateProperty
+            = BindableProperty.Create(nameof(State), typeof(SideMenuState), typeof(SideMenuView), SideMenuState.Default, BindingMode.TwoWay, propertyChanged: OnStatePropertyChanged);
+
+        public static readonly BindableProperty CurrentGestureStateProperty
+            = BindableProperty.Create(nameof(CurrentGestureState), typeof(SideMenuState), typeof(SideMenuView), SideMenuState.Default, BindingMode.OneWayToSource);
+
+        public static readonly BindableProperty PositionProperty
+            = BindableProperty.CreateAttached(nameof(GetPosition), typeof(SideMenuPosition), typeof(SideMenuView), SideMenuPosition.None);
+
+        public static readonly BindableProperty MenuWidthPercentageProperty
+            = BindableProperty.CreateAttached(nameof(GetMenuWidthPercentage), typeof(double), typeof(SideMenuView), -1.0);
+
+        public static readonly BindableProperty MenuGestureEnabledProperty
+            = BindableProperty.CreateAttached(nameof(GetMenuGestureEnabled), typeof(bool), typeof(SideMenuView), true);
+
+        readonly List<TimeDiffItem> timeDiffItems;
+
+        readonly SideMenuElementCollection children;
+
+        AbsoluteLayout mainLayout;
```

Comment:
Does it make sense to have all the private stuff in the same place above?

Reply:
Do you mean to put it near constants?